primefaces/src/main/java/org/primefaces/model/DefaultLazyDataModel.java

```diff
@@ -68,19 +68,7 @@ public List<T> load(int first, int pageSize, Map<String, SortMeta> sortBy, Map<S
 
         sort(filteredValues);
 
-        if (pageSize == 0) {
-            return filteredValues;
-        }
-
-        int last = first + pageSize;
-        if (last > filteredValues.size()) {
-            last = filteredValues.size();
-        }
-        if (first > last) {
-            first = 0;
-        }
```

Comment:
Not sure when that can happen

Reply:
If data is deleted in meantime?

Reply:
Yeah but that case is already handled by UIPageableData#calculateFirst in DataTable#loadLazyScrollData

Reply:
Maybe already fixed in meantime
@jungm maybe you can retest

Reply:
@Rapster that was a fix with jungm see: Fix #11521: Datatable first value fix (#11529)

Reply:
This was my original issue that started all of this: https://github.com/primefaces/primefaces/issues/11401

Fixed it in https://github.com/primefaces/primefaces/pull/11402 in a slightly different way. I also added an IT for this exact issue when I fixed it, if it still passes everything should be good for me :slightly_smiling_face:

#11521 moved this to another place because it caused issues for other users

Reply:
ok, the IS is passing anyway and if it's not failing it's certainly because skip/limit can handle the job